### PR TITLE
Gwf member revamp

### PR DESCRIPTION
**Pull request by garyfoster**

### CODE REVIEW

File: app/controllers/concerns/radbear_members_controller.rb [hunk 1]
```diff
@@ -2,16 +2,23 @@ module RadbearMembersController
...
 
   def index
-    @members = Member.authorized(current_member).recent_first.page(params[:page])
+    @members = Member.authorized(current_member)
+    @pending = @members.pending.by_id
+    @members = @members.recent_first.page(params[:page])
 
-    if params[:status]
-      @status = params[:status]
+        if params[:status]
```
Comment (plicjo) [inline app/controllers/concerns/radbear_members_controller.rb]: This spacing is a bit jacked up.

File: app/models/concerns/radbear_company.rb [hunk 2]
```diff
@@ -95,8 +95,8 @@ def active_member_for_name(first_name, last_name)
     return self.members_for_name(first_name, last_name).active.limit(1).first
   end
 
-  def add_member(user, security_group, first_name, last_name, approved, client)
-    params = {user: user, approved: approved, security_group: security_group, first_name: first_name, last_name: last_name}
+  def add_member(user, security_group, first_name, last_name, member_status, client)
```
Comment (plicjo) [inline app/models/concerns/radbear_company.rb]: I see that you updated the seeds in the Groundswell PR. All RBMT apps will need to do the same.

File: app/models/concerns/radbear_member.rb [hunk 3]
```diff
@@ -45,27 +46,7 @@ def admin
   end
 
   def active
-    return self.status == "Active"
-  end
-
-  def status
```
Comment (plicjo) [inline app/models/concerns/radbear_member.rb]: Hot refactor

File: app/views/members/index.html.haml [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
-          %a{class: "btn dropdown-toggle #{@status == "Approved" ? 'btn-success' : 'btn-warning'}", "data-toggle" => "dropdown", href: "#"}
-            = @status + " Members"
+          %a{class: "btn dropdown-toggle #{@status.nil? ? "btn-warning" : @status.button_style}", "data-toggle" => "dropdown", href: "#"}
+            = @status.nil? ? "All Members" : @status.name + " Members"
             %span.caret
           %ul.dropdown-menu
-            %li= link_to "Approved", members_path(status: "Approved")
-            %li= link_to "Unapproved", members_path(status: "Unapproved")
-            %li= link_to "All", members_path(status: "All")
+            -@member_statuses.each do |status|
```
Comment (plicjo) [inline app/views/members/index.html.haml]: This is nice.

File: app/models/concerns/radbear_member.rb [hunk 5]
```diff
@@ -80,29 +61,14 @@ def validate_member
       end
     end
 
-    def validate_group
```
Comment (plicjo) [inline app/models/concerns/radbear_member.rb]: Do we still need this check?

File: app/views/companies/show.html.haml [hunk 6]
```diff
@@ -12,14 +12,16 @@
             %tr
               %th Name
               %th Email
-              %th Status
+              %th Member Status
```
Comment (plicjo) [inline app/views/companies/show.html.haml]: Will "Member Status" vs "User Status" be confusing to the end user?

File: app/controllers/concerns/radbear_members_controller.rb (hunk 1 above)
Comment (garyfoster) [inline app/controllers/concerns/radbear_members_controller.rb]: yep, fixing

File: app/models/concerns/radbear_company.rb (hunk 2 above)
Comment (garyfoster) [inline app/models/concerns/radbear_company.rb]: ya, i have a punch list that i'm adding a task for - there is currently only one other RBMT application so it's not too hard to manage

File: app/models/concerns/radbear_member.rb (hunk 5 above)
Comment (garyfoster) [inline app/models/concerns/radbear_member.rb]: no, it's no longer applicable

File: app/views/companies/show.html.haml (hunk 6 above)
Comment (garyfoster) [inline app/views/companies/show.html.haml]: probably? I was wondering about that but not sure how to abstract it - I think we'll want to look at that again one day unless you have any suggestions. currently it only shows a value in user status when it's something other than normal like they haven't confirmed their email, or haven't accepted the invite

**Pull request merged**